usr/share/rear/lib/layout-functions.sh

```diff
@@ -1182,11 +1203,18 @@ delete_dummy_partitions_and_resize_real_ones() {
 
     # Resize previously shrinked partitions (to make place for dummy
     # partitions) to expected size
+    local -i startB
     local -i endB
-    while read num endB ; do
+    while read num startB endB ; do
         LogPrint "Disk '$current_disk': resizing partition number $num to original size"
-        parted -s -m $current_disk resizepart $num "${endB}B"
-    done <<< "$(printf "%d %d\n" "${partitions_to_resize[@]}")"
+        if is_true $FEATURE_PARTED_RESIZEPART ; then
+            parted -s -m $current_disk resizepart $num "${endB}B"
+        elif is_true $FEATURE_PARTED_RESIZE_NEED_START ; then
+            parted -s -m $current_disk resize $num "${startB}B" "${endB}B"
+        else
```

Comment:
This `else` clause here would be done even if `FEATURE_PARTED_RESIZE=n`
and currently I don't see that delete_dummy_partitions_and_resize_real_ones
is not called if `FEATURE_PARTED_RESIZE=n`.

To be more on the safe side I would perefer when the code is locally fail safe like
```
        elif is_true FEATURE_PARTED_RESIZE ; then
            parted -s -m $current_disk resize $num "${endB}B"
        else
            BugError "delete_dummy_partitions_and_resize_real_ones called but parted cannot resize"
        fi
```
or at least an explanatory comment how things work as intended
even in case of `FEATURE_PARTED_RESIZE=n`.